source/agora/network/Manager.d

```diff
@@ -994,25 +995,42 @@ public class NetworkManager
     public void getUnknownTXs (Ledger ledger) @safe nothrow
     {
         auto unknown_txs = ledger.getUnknownTXHashes();
+        log.trace("getUnknownTXs: detected {} unknown txs", unknown_txs.length);
 
         foreach (peer; this.peers[])
         {
             if (unknown_txs.length == 0)
                 break;
 
-            foreach (tx; peer.client.getTransactions(unknown_txs))
+            // Fetch transactions in chunks
+            auto hashes = unknown_txs[].map!((Hash h) => h).array; // can not be lazy so use array
+            auto added = hashes.chunks(8)
+                .map!(chunk => Set!Hash.from(chunk))
+                .map!(hash_chunk => peer.client.getTransactions(hash_chunk))
+                .map!(txs => addTxs(ledger, txs))
+                .sum();
+            log.trace("getUnknownTXs: Added {} txs to tx pool", added);
+        }
+    }
+
+    // Add the chunk of txs fetched from the other node
+    private size_t addTxs (Ledger ledger, Transaction[] txs) @safe nothrow
+    {
+        auto accepted = 0;
+        txs.each!((tx)
+        {
+            try
             {
-                try
-                {
-                    ledger.acceptTransaction(tx);
-                    unknown_txs.remove(tx.hashFull());
```

Comment:
This is still needed, so that we don't fetch same TXs from different peers over and over again.

Reply:
It is removed in the `ledger.acceptTransaction`.

Reply:
Yeap, but that does not affect our local `unknown_txs` here. The for loop in `getUnknownTXs` iterating over the peers would request same TXs even if we got it from one of the previous peers.

Reply:
Well spotted. I'll update.

Reply:
Now it refreshes the set after each peer.